public/app/features/transformers/editors/EnumMappingRow.tsx

```diff
@@ -0,0 +1,160 @@
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputRef]);
+
+  const onEnumInputChange = (event: FormEvent<HTMLInputElement>) => {
+    if (
+      event.currentTarget.value !== '' &&
+      checkIsEnumUniqueValue(event.currentTarget.value) &&
+      event.currentTarget.value !== value
+    ) {
+      setValidationError('Enum value already exists');
+    } else {
+      setValidationError(null);
+    }
+
+    setEnumValue(event.currentTarget.value);
+  };
+
+  const onEnumInputBlur = () => {
+    setIsEditing(false);
+    setValidationError(null);
+
+    // Do not add empty or duplicate enum values
+    if (enumValue === '' || validationError !== null) {
+      onRemoveEnumRow(mappedIndex);
+      return;
+    }
+
+    onChangeEnumValue(mappedIndex, enumValue);
+  };
+
+  const onEnumInputKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      onEnumInputBlur();
+    }
+  };
+
+  const onEnumValueClick = () => {
+    setIsEditing(true);
+  };
+
+  const onRemoveButtonClick = () => {
+    onRemoveEnumRow(mappedIndex);
+  };
+
+  return (
+    <Draggable
+      key={`${convertFieldTransformIndex}/${value}`}
+      draggableId={`${convertFieldTransformIndex}/${value}`}
+      index={index}
+    >
+      {(provided) => (
+        <tr key={index} ref={provided.innerRef} {...provided.draggableProps}>
+          <td>
+            <div className={styles.dragHandle} {...provided.dragHandleProps}>
+              <Icon name="draggabledots" size="lg" />
+            </div>
+          </td>
+          {isEditing ? (
+            <td>
+              <Input
+                ref={inputRef}
+                type="text"
+                value={enumValue}
+                onChange={onEnumInputChange}
+                onBlur={onEnumInputBlur}
+                onKeyDown={onEnumInputKeyDown}
+              />
+              {validationError && <FieldValidationMessage>{validationError}</FieldValidationMessage>}
+            </td>
+          ) : (
+            <td onClick={onEnumValueClick} className={styles.clickableTableCell}>
+              {value && value !== '' ? value : 'Click to edit'}
+            </td>
+          )}
+          <td className={styles.textAlignCenter}>
+            <HorizontalGroup spacing="sm">
+              <IconButton
+                name="trash-alt"
+                onClick={onRemoveButtonClick}
+                data-testid="remove-enum-row"
+                aria-label="Delete enum row"
+                tooltip="Delete"
+              />
+            </HorizontalGroup>
+          </td>
+        </tr>
+      )}
+    </Draggable>
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  dragHandle: css({
+    cursor: 'grab',
+    // create focus ring around the whole row when the drag handle is tab-focused
+    // needs position: relative on the drag row to work correctly
+    '&:focus-visible&:after': {
```

Comment:
i think technically it's `::after`, but :shrug: 

https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Selectors/Pseudo-classes_and_pseudo-elements

Reply:
turns out this is just copy pasta from the value mappings draggable implementation / not actually doing anything here 😬